lib/provider.coffee

```diff
@@ -22,13 +18,18 @@ class Provider
   # the default suggestions or should be displayed exclusively
   exclusive: false
 
+  buildSuggestionsShim: (options) ->
+    return unless options?.editor?
+    @editor = options.editor
+    @buildSuggestions.apply(this, arguments)
+
   # Public: Gets called when the document has been changed. Returns an array with
   # suggestions. If `exclusive` is set to true and this method returns suggestions,
   # the suggestions will be the only ones that are displayed.
   #
   # Returns An {Array} of suggestions.
   buildSuggestions: ->
-    throw new Error "Subclass must implement a buildWordList(prefix) method"
+    throw new Error 'Subclass must implement a buildWordList(prefix) method'
```

Comment:
We already pass an options object to this method. This method is legacy and will be deprecated.